.github/workflows/test_head_and_bump.yaml

```diff
@@ -27,19 +27,21 @@ jobs:
         run: |
           python $GITHUB_WORKSPACE/build_tools/configure.py
           python3 -m pip install . -f https://github.com/google/iree/releases --force-reinstall .[xla,cpu,test]
-      
+
       - name: Run Tests
         run: |
           build_tools/unittest.sh
 
       - name: Update Pinned Versions
         run: |
-          python build_tools/update_pinned.py
+          IREE_VERSION=$(python build_tools/update_pinned.py)
           cat version_info.json
-
-      - name: Push updated versions
-        run: |
           git config --local user.name "Bump Pinned Versions"
           git config --local user.email "iree-github-actions-bot@google.com"
-          git commit -am "Update pinned versions"
-          git push origin update-versions
+          git commit -am "Update IREE to ${IREE_VERSION}"
+
+      - name: Pushing changes
+        run: |
+          git remote set-url origin https://x-access-token:${{ secrets.GITHUB_TOKEN }}@github.com/${{ github.repository }}
```

Comment:
Checkout should have handled all this. It persists credentials by default (that's why I was saying that a fancy action wasn't necessary)

Reply:
Removed.